examples/nextjs-typescript/components/NextChakraLink.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { PropsWithChildren } from "react"
+import NextLink from "next/link"
+import { LinkProps as NextLinkProps } from "next/dist/client/link"
+import {
+  Link as ChakraLink,
+  LinkProps as ChakraLinkProps,
+} from "@chakra-ui/core"
+
+export type NextChakraLinkProps = PropsWithChildren<
```

Comment:
This type should include `& Omit<NextLinkProps, "passHref">` because it's hard-coded to `true` and it's not a valid prop to pass to `ChakraLink`.